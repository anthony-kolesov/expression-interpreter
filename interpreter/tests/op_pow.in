# Typical operation.
out 5 ^ 3
print "\n"

# Relatively big numbers.
out 10000 ^ 2
print "\n"

# Crossing the zero.
out -4 ^ 4
print "\n"

# Staying negative.
out -4 ^ 3
print "\n"

# Floating point numbers.
out 4.0 ^ 1.5
print "\n"

# Zero values.
out 2 ^ 0
print "\n"

out 0 ^ 2
print "\n"

out 0 ^ 0
print "\n"

# By one.
out 4 ^ 1
print "\n"
