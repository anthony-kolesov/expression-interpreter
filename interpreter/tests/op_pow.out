125
100000000
256
-64
8.000000
1
0
1
4
